src/librustdoc/html/layout.rs

```diff
@@ -197,7 +198,11 @@ pub fn render<T: fmt::Display, S: fmt::Display>(
     title     = page.title,
     description = page.description,
     keywords = page.keywords,
-    favicon   = if layout.favicon.is_empty() {
+    favicon   = if layout.favicon_path.is_some() {
+        format!(r#"<link rel="shortcut icon" href="{static_root_path}{path}">"#,
+                static_root_path=static_root_path,
+                path=layout.favicon)
```

Comment:
This doesn't seem like a useful addition to the layout code. If we're going to use the `favicon` string and the `static_root_path` anyway, why not just update that path with the right path?

(If we make the favicon path a plain string instead of an actual file path, we can keep this distinction, though, as long as we make sure to *not* use `static_root_path` in that situation.)

Reply:
Because one generates a full path whereas the other generates a relative one.